source/slang/memory_pool.cpp

```diff
@@ -1,52 +0,0 @@
-#include "memory_pool.h"
-
-namespace Slang
-{
-    const size_t kPoolSegmentSize = 4 << 20; // use 4MB segments
```

Comment:
Wow. That sure looks like code I wrote, so I have no way to dodge the blame for the 4MB segment size. I wonder what in the world I was thinking.

Anyway, it is good to consolidate on a single implementation of scoped memory allocation. It would be good if we can start to corral more of the compiler's memory usage similar to what we are doing for the IR (there is no strong reason why the AST should be as destructor-happy as it is...).